ouroboros-consensus/docs/GitProcess.md

```diff
@@ -0,0 +1,79 @@
+# Consensus git process
+
+This document describes the git history we strive for in the Consensus Team. The
+git history is a key asset for understanding and for debugging, so we take care
+to refine it before merging each PR.
+
+## Guiding principles
+
+_Messages matter_: Commit messages summarize an atomic coherent change and its
+motivation. They help reviewers do their job, future developers understand the
+past design process, troubleshooters debug changes in behavior, downstream
+developers understand recent changes and their context, and so on. In these kind
+of scenarios, commit messages are often the easiest documentation to read first.
+
+_Small and organized commits_: The commits themselves, beyond just their
+messages, also contribute to the story in their own way. Prefer small commits
+that include only necessary changes, and choose a commit order that seems most
+natural to you. Doing so often also makes it easier to write useful commit
+messages. Rebase before merging, so the repository's full story is as linear as
+it can be. At the very least, this effort will make it easier for your teammates
+to review the PR.
+
+_Never break the build_: The code should build and pass all test suites after
+each commit, not just after each PR. There is no good reason for a commit to
+break the build. It is only slightly less important that the tests pass after
+each commit. The Continuous Integration (CI) machinery ensures that the tip of a
+PR builds before it's merged, so every commit on the `master` branch will build
+and pass the tests that are enabled in CI. In particular, this process ensures
+`git bisect` can be as useful as possible -- when you reach for that kind of
+sledgehammer, you want it to work well.
+
+Note that CI enables `-Werror`, but we don't require you to enable it locally.
+If you'd like to opt-in to that or similar, add a `cabal.project.local` file.
+
+## Guidelines
+
+  * Write useful commit messages. There are lots of guides on the Internet about
```

Comment:
It's not a scope of this PR but maybe it is worth exploring some form of standardization. I found this blog post very useful years back https://chris.beams.io/posts/git-commit/

Reply:
That's a good article, yeah. In my experience, if it's legible and the content is taken seriously, that's enough. I think that's basically two of the seven rules in that article.

But obviously I'm not against codifying practices as rules ;). We just haven't prioritized them for commit messages yet.

Reply:
I opened Issue #2914.